src/components/principal-bill-info-form.jsx

```diff
@@ -1,24 +1,112 @@
 import React from "react";
-import './styles/principal-bill-info-form.css'
+import './styles/forms.css'
+import { Link } from "react-router-dom";
+import swal from 'sweetalert';
 
 class PrincipalBillInfoForm extends React.Component {
+    // myStorage = window.sessionStorage;
+    startBillingDate
+    endBillingDate
+    discounts
+    discountsValue
+    cleaning
+    cleaningValue
+    button
+    today
+    componentDidMount() {
+        this.startBillingDate = document.getElementById('start-billing-date')
+        this.endBillingDate = document.getElementById('end-billing-date')
+        this.discounts = document.getElementById('discounts')
+        this.discountsValue = this.discounts.value
+        this.cleaningValue = this.discounts.value
+        this.cleaning = document.getElementById('cleaning')
+        this.button = document.getElementById('principalBillInfoBtn')
+        this.today = new Date();
+
+
+        var dd = this.today.getDate();
+        var mm = this.today.getMonth() + 1;
+        var yyyy = this.today.getFullYear();
+
+        if (dd < 10) {
+            dd = '0' + dd;
+        }
+        if (mm < 10) {
+            mm = '0' + mm;
+        }
+        this.today = yyyy + '-' + mm + '-' + dd;
+
+        this.startBillingDate.setAttribute("max", this.today);
+        this.endBillingDate.setAttribute("max", this.today);
+        sessionStorage.clear();
+    }
+
+
+    handleChange = () => {
+        if (this.startBillingDate.value >= this.endBillingDate.value) {
+            swal({
+                title: "Error",
+                text: "La fecha final debe ser mayor a la fecha inicial",
+                icon: "warning",
+                button: "Continuar",
+            });
+            this.startBillingDate.value = ''
+        }
+        if (this.endBillingDate.value > this.today) {
+            swal({
+                title: "Error",
+                text: "Fecha invalida",
+                icon: "warning",
+                button: "Continuar",
+            });
+            this.endBillingDate.value = ''
+        }
+        if (this.startBillingDate.value === "" || this.endBillingDate.value === '' || this.discounts.value === '' || this.cleaning.value === '') {
```

Comment:
Crear una variable por fuera del if para que ese condicional tan largo se pueda entender en lenguaje natural, por ejemplo 
`const isFormNotEmpty = this.startBillingDate.value === "" || this.endBillingDate.value === '' || this.discounts.value === '' || this.cleaning.value === ''

if (isFormNotEmpty) {
 ....
}`